tiledb/api/c_api/filter/filter_api_experimental.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file tiledb/api/c_api/filter/filter_api_experimental.h
```

Comment:
Why would we like those to go in `experimental`? 

And that's also a question for the whole team, I am not aware of the strategy around `experimental` APIs, when do we label one as such and when we consider it mature enough to get it out of that state?